src/components/table-toolbar/KeycloakDataTable.tsx

```diff
@@ -242,26 +242,58 @@ export function KeycloakDataTable<T>({
   );
 
   const _onSelect = (isSelected: boolean, rowIndex: number) => {
+
     if (rowIndex === -1) {
-      setRows(
-        rows!.map((row) => {
-          row.selected = isSelected;
-          return row;
-        })
-      );
+
+      if (filteredData) {
+        setRows(
+          filteredData!.map((row) => {
+            row.selected = isSelected;
+            return row;
+          })
+        );
+      } else {
+        setRows(
+          rows!.map((row) => {
+            row.selected = isSelected;
+            return row;
+          })
+        );
+        setRows(rows);
+      }
     } else {
-      rows![rowIndex].selected = isSelected;
+      if (filteredData) {
+        const foundRow = rows!.find(
+          (row) => row.data === filteredData[rowIndex].data
+        );
+
+        filteredData[rowIndex].selected = isSelected;
+
+        if (foundRow) {
+          foundRow.selected = isSelected;
+        }
+        setRows([...filteredData!]);
+      } else {
+        rows![rowIndex].selected = isSelected;
+      }
+
```

Comment:
Same here? Could it be replaced by:
```suggestion
      (filteredData || rows)[rowIndex].selected = isSelected;
```

Reply:
Thank you for your helpful suggestions @edewit ! I tested this one out - if we make this change, the selected items do not remain selected if a user enters a search filter (all the selected items will get reset).